### PR TITLE
Data processing

### PR DESCRIPTION
**Pull request by Thoth1111**
In this branch I achieved the following: 
✅ Create model classes for all entities in the ERD diagram
✅ Setup associations between the models
✅ Test CRUD methods on entities
✅ Create a custom method in `User` that returns up to 3 latest posts of a given user.
✅ Create a custom method in `Post` to update the posts counter for a user.
✅ Create a custom method in `Post` that returns up to 5 latest comments of a user.
✅ Create a custom method in `Comment` to update the comments counter for a given post.
✅ Create a custom method in `Like` to update the likes counter for a given post.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: :author_id, dependent: :destroy
+  has_many :comments, through: :posts
+  has_many :likes, through: :posts
```
Comment (rloterh) [inline app/models/user.rb]: - [x] With the given [_ERD_](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png), whiles establishing a _has many_ **association** for _User_ model, _author_id_ (_Author_ is aliasing _User_ model) should be referenced as **foreign key** in _Comment,_ and _Like_ models. Kindly ensure to improve upon this aspect, and our code should be similar to:
```ruby
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
```
Comment (rloterh) [inline app/models/post.rb]: - [x] The `id` of _Post_ is supposed to reference the foreign key `author_id` (declared in the _User_ model) of the _User_ model, and hence it's redundant to declare `author_id` as foreign key in _Post_ model. Simply do something like:
```ruby
# app/models/post.rb
belongs_to :author, class_name: 'User'

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class Comment < ApplicationRecord
+  belongs_to :post
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
```
Comment (rloterh) [inline app/models/comment.rb]: - [x] Similarly, Comment's `id` should reference the foreign key `author_id` (declared in the _User_ model) of the _User_model, and hence it's redundant to declare `author_id` as foreign key in _Comment_ model. Simply do something like:
```ruby
# app/models/comment.rb
belongs_to :author, class_name: 'User'

File: app/models/like.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
```
Comment (rloterh) [inline app/models/like.rb]: - [x] Again, Like's`id` should reference the foreign key `author_id` (declared in the _User_ model) of the _User_ model, and hence it's redundant to declare `author_id` as foreign key in _Like_ model. Simply do something like:
```ruby
# app/models/like.rb
belongs_to :author, class_name: 'User'

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Alwyn, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🏁 🚦  🚗 _ 🏃 🌴  🏠 🌳 🏘️ 🏡  

## Highlights 👍  
- Good job so far with creating model associations!
- ReadMe looks great!

## Required Changes ♻️
- [x] MIT License link in the ReadMe is broken, kindly ensure to fix it.

_Kindly also check the other inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Thoth1111** [comment]
Thank you. All changes have been addressed in the recent commits.

**Review by afanimartin — APPROVED** [review]
Hi @Thoth1111, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- MIT license link fixed
- Associations updated as requested

# Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

## Find me on slack
### Martin Afani

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**